tensorboard/plugins/histogram/summary.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Histogram summaries and TensorFlow operations to create them.
+
+A histogram summary stores a list of buckets. Each bucket is encoded as
+a triple `[left_edge, right_edge, count]`. Thus, a full histogram is
+encoded as a tensor of dimension `[k, 3]`.
+
+In general, the value of `k` (the number of buckets) will be a constant,
+like 30. There are two edge cases: if there is no data, then there are
+no buckets (the shape is `[0, 3]`); and if there is data but all points
+have the same value, then there is one bucket whose left and right
+endpoints are the same (the shape is `[1, 3]`).
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+
+import tensorflow as tf
+import numpy as np
+
+from tensorboard.plugins.histogram import metadata
+
+
+BUCKETS = 30
+
+
+def _buckets(data, bucket_count=None):
+  """Create a TensorFlow op to group data into histogram buckets.
+
+  Arguments:
+    data: A `Tensor` of any shape. Must be compatible with `float64`.
+    bucket_count: Optional positive `int` or scalar `int32` `Tensor`.
+  Returns:
+    A `Tensor` of shape `[k, 3]` and type `float64`. The `i`th row is
+    a triple `[left_edge, right_edge, count]` for a single bucket.
+    The value of `k` is either `bucket_count` or `1` or `0`.
+  """
+  if bucket_count is None:
+    bucket_count = BUCKETS
+  known_nonpositive = False
+  if tf.is_numeric_tensor(bucket_count):
```

Comment:
This junk works around https://gist.github.com/wchargin/6b1fb62119d7190b1301c6ba8a401e78, and is what apassos suggested to use. It's not great in that it introduces a contrib dependency and is somewhat messy code, so I'd be okay with removing this assertion entirely if you want.